### DISCUSSION TITLE
How to activate cache for variables

### DISCUSSION THREAD

**Discussion by gwlucastrig** [Unidata/netcdf-java#1399] [Questions and Answers] [answered]
I am processing a NetCDF file containing a Variable with dimensions 10800 by 21600, and chunk size 1350U, 2700U.  I wish to scan it in row-major order.   It works like a charm when I activate the cache:

`z.setCaching(true);`  

But if I do not make this call, the process absolutely crawls (rather than taking 10 seconds, it takes nearly an hour) .   Well, using setCaching() is awesome, except that the method is deprecated.  As far as I can tell, there is no other way to enable the cache.   I tried:

`z.createNewCache();`
 
which does _not_ make any difference. 

So, my question is what am I missing and what is the recommended way to proceed?

Thanks.

P.S.  I am using cdm-core version 5.6.0.  The data file in question is the ETOPO Global Relief Model available at the [NCEI THREADDS Server](https://www.ngdc.noaa.gov/thredds/catalog/global/ETOPO2022/60s/60s_surface_elev_netcdf/catalog.html?dataset=globalDatasetScan/ETOPO2022/60s/60s_surface_elev_netcdf/ETOPO_2022_v1_60s_N90W180_surface.nc)

**Comment by lesserwhirls** [comment, marked as answer]
Greetings @gwlucastrig! That deprecation was added a few years back when we were looking to update the API to use produce immutable objects. I'm still playing a bit of catchup since returning to Unidata to see where those efforts landed. However, even in light of immutability, being able to call `setCaching` for the read path is important, as you have pointed out. ~I consider that method as being deprecated for use outside of reading (not sure what that use-case would be at the moment), and I can update the docstring to reflect that.~ I'm going to need to give that more thought. There will be a way to allow cached reads on a per variable basis, but I'm not quite sure what it will look like at this point. I need to do some digging to see if there are any indications as to what the path forward was with regards to the API rewrite.

**Reply by gwlucastrig** [reply]
Thanks for the information. I'm grateful to the Unidata team for the work you do and that this issue has come to light.
